ddtrace/contrib/asyncio/tracer.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+
+from ...tracer import Tracer
+from ...context import Context
+
+
+class AsyncContextMixin(object):
+    """
+    Defines by composition how to retrieve the ``Context`` object, while
+    running the tracer in an asynchronous mode with ``asyncio``.
+    """
+    def get_call_context(self, loop=None):
+        """
+        Returns the scoped Context for this execution flow. The ``Context`` uses
+        the current task as a carrier so if a single task is used for the entire application,
+        the context must be handled separately.
+        """
+        # TODO: this may raise exceptions; provide defaults or
+        # gracefully "log" errors
+        loop = loop or asyncio.get_event_loop()
+
+        # the current unit of work (if tasks are used)
+        task = asyncio.Task.current_task(loop=loop)
+        if task is None:
+            # FIXME: this will not work properly in all cases
+            # if the Task is None, the application will crash with unhandled exception
+            # if we return a Context(), we will attach the trace to a (probably) wrong Context
+            return
+        try:
+            # return the active Context for this task (if any)
+            return task.__datadog_context
+        except (KeyError, AttributeError):
```

Comment:
👍 I will fix all these nitpicks because I didn't pay attention to a lot of stuff while achieving the async compatibility.